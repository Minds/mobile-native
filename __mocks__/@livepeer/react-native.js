// mock @livepeer/react-native
export const LivepeerConfig = {
  setConfig: jest.fn(),
  getConfig: jest.fn(),
  get: jest.fn(),
  set: jest.fn(),
};

export const createReactClient = jest.fn();
export const createLivepeerClient = jest.fn();
export const Player = jest.fn();
