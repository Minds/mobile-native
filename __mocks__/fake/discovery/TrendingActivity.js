export const TrendingActivity = {
  id: '1273869708835164175',
  entity: {
    guid: '1273869708835164175',
    type: 'activity',
    time_created: Date.now() / 1000 - 3200,
    time_updated: Date.now() / 1000 - 3200,
    container_guid: '1240157798176661518',
    owner_guid: '1240157798176661518',
    access_id: '2',
    tags: ['Nature'],
    nsfw: [],
    nsfw_lock: [],
    allow_comments: true,
    title: false,
    blurb: false,
    perma_url: false,
    message: "Wow, what do we call this? It's beautiful, isn't it?\n#Nature",
    ownerObj: {
      guid: '1240157798176661518',
      type: 'user',
      subtype: false,
      time_created: '1621052660',
      time_updated: false,
      container_guid: '0',
      owner_guid: '0',
      site_guid: false,
      access_id: '2',
      tags: ['myphoto', 'outdoors', 'news', 'minds'],
      nsfw: [],
      nsfw_lock: [],
      allow_comments: false,
      name: 'SIMPLEBEST1996',
      username: 'simplebest1996',
      language: 'en',
      icontime: '1628272389',
      legacy_guid: false,
      featured_id: false,
      banned: 'no',
      ban_reason: false,
      website: false,
      briefdescription: "Subscribe to my channel and I'll do same!!",
      gender: false,
      city: false,
      merchant: false,
      boostProPlus: false,
      fb: false,
      mature: 0,
      monetized: false,
      signup_method: false,
      social_profiles: [],
      feature_flags: false,
      programs: [],
      plus: false,
      hashtags: false,
      verified: false,
      founder: false,
      disabled_boost: false,
      boost_autorotate: true,
      categories: [],
      wire_rewards: null,
      pinned_posts: [],
      is_mature: false,
      mature_lock: false,
      last_accepted_tos: 1558597098,
      opted_in_hashtags: 4,
      last_avatar_upload: '1628272389',
      canary: false,
      theme: 'dark',
      toaster_notifications: true,
      mode: 0,
      btc_address: '',
      surge_token: '',
      hide_share_buttons: false,
      allow_unsubscribed_contact: false,
      dismissed_widgets: [],
      liquidity_spot_opt_out: 0,
      'thumbs:up:count': '0',
      'thumbs:down:count': '0',
      'thumbs:up:user_guids': [],
      'thumbs:down:user_guids': [],
      urn: 'urn:user:1240157798176661518',
      subscribed: false,
      subscriber: false,
      boost_rating: 1,
      pro: false,
      pro_published: false,
      rewards: true,
      p2p_media_enabled: false,
      is_admin: false,
      onchain_booster: 0,
      email_confirmed: true,
      eth_wallet: '',
      rating: 1,
      disable_autoplay_videos: false,
      yt_channels: [],
    },
    containerObj: false,
    thumbnail_src:
      'https://cdn.minds.com/fs/v1/thumbnail/1273869385403994122/xlarge/',
    remind_object: false,
    entity_guid: '1273869385403994122',
    featured: false,
    featured_guid: false,
    custom_type: 'batch',
    custom_data: [
      {
        src:
          'https://cdn.minds.com/fs/v1/thumbnail/1273869385403994122/xlarge/',
        href:
          'https://www.minds.com/media/1240157798176661518/1273869385403994122',
        mature: false,
        width: 819,
        height: 1024,
        gif: false,
      },
    ],
    'thumbs:up:count': 14,
    'thumbs:up:user_guids': [
      '1240157798176661518',
      '1239934369339023374',
      '1235147474172125192',
      '1239654407470063619',
      '1239481325111484434',
      '1239689805399334929',
      '1239506512825557007',
      '1270017288434618374',
      '1241276369787166740',
      '1239994569018318857',
      '1257313820917899274',
      '1235155738494509066',
      '1239673036664217604',
      '1235148984738127874',
    ],
    'thumbs:down:count': 0,
    'thumbs:down:user_guids': [],
    p2p_boosted: false,
    mature: false,
    monetized: false,
    paywall: '',
    edited: '',
    comments_enabled: true,
    wire_totals: false,
    boost_rejection_reason: -1,
    pending: '',
    rating: 2,
    ephemeral: false,
    hide_impressions: false,
    pinned: false,
    time_sent: 1629090205,
    permaweb_id: '',
    'comments:count': 5,
    urn: 'urn:activity:1273869708835164175',
    impressions: 30,
    reminds: 11,
    wire_threshold: null,
    license: '',
    thumbnails: {
      xlarge:
        'https://cdn.minds.com/fs/v1/thumbnail/1273869385403994122/xlarge/',
      large: 'https://cdn.minds.com/fs/v1/thumbnail/1273869385403994122/large/',
    },
  },
  guid: '1273869708835164175',
  hashtag: 'nature',
  title: "Wow, what do we call this? It's beautiful, isn't it?\n#Nature",
  volume: 5,
  period: 9.224444444444444,
  selected: null,
};
