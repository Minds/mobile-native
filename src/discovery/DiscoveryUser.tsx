//@ts-nocheck
import React, { Component } from 'react';

import {
  TouchableOpacity,
  Image,
  StyleSheet,
  Keyboard,
  Text,
  View,
} from 'react-native';

import { observer } from 'mobx-react';

import { MINDS_CDN_URI } from '../config/Config';

import { CommonStyle } from '../styles/Common';
import { FLAG_SUBSCRIBE, FLAG_VIEW } from '../common/Permissions';
import SubscriptionButton from '../channel/subscription/SubscriptionButton';
import ThemedStyles from '../styles/ThemedStyles';

type PropsType = {
  row: any;
};

type StateType = {
  guid: string | null;
};

@observer
class DiscoveryUser<T extends PropsType> extends Component<T, StateType> {
  /**
   * State
   */
  state = {
    guid: null,
  };

  /**
   * Derive state from props
   * @param {object} nextProps
   * @param {object} prevState
   */
  static getDerivedStateFromProps(nextProps, prevState) {
    if (prevState.guid !== nextProps.row.item.guid) {
      return {
        guid: nextProps.row.item.guid,
        source: {
          uri: MINDS_CDN_URI + 'icon/' + nextProps.row.item.guid + '/medium',
        },
      };
    }

    return null;
  }

  /**
   * Navigate To channel
   */
  _navToChannel = () => {
    Keyboard.dismiss();
    if (this.props.navigation) {
      if (
        this.props.row.item.isOpen() &&
        !this.props.row.item.can(FLAG_VIEW, true)
      ) {
        return;
      }
      this.props.navigation.push('Channel', { entity: this.props.row.item });
    }
  };

  /**
   * Render right button
   */
  renderRightButton() {
    const channel = this.props.row.item;

    if (
      channel.isOwner() ||
      this.props.hideButtons ||
      (channel.isOpen() && !channel.can(FLAG_SUBSCRIBE))
    ) {
      return;
    }

    const testID = this.props.testID
      ? `${this.props.testID}SubscriptionButton`
      : 'subscriptionButton';

    return <SubscriptionButton channel={channel} testID={testID} />;
  }

  /**
   * Render
   */
  render() {
    const { row, ...otherProps } = this.props;

    return (
      <TouchableOpacity
        style={[
          styles.row,
          ThemedStyles.style.borderPrimary,
          ThemedStyles.style.borderBottomHair,
        ]}
        onPress={this._navToChannel}
        {...otherProps}>
        <Image source={this.state.source} style={styles.avatar} />
        <View style={[CommonStyle.flexContainerCenter]}>
          <Text style={[styles.body, CommonStyle.fontXL]}>{row.item.name}</Text>
          <Text
            style={[styles.body, CommonStyle.fontS, CommonStyle.colorMedium]}>
            @{row.item.username}
          </Text>
        </View>
        {this.renderRightButton()}
      </TouchableOpacity>
    );
  }
}

export default DiscoveryUser;

const styles = {
  row: {
    flexDirection: 'row',
    alignItems: 'center',
    flexWrap: 'wrap',
    paddingTop: 10,
    paddingLeft: 12,
    paddingBottom: 10,
    paddingRight: 12,
  },
  body: {
    marginLeft: 8,
    height: 22,
    // flex: 1,
  },
  avatar: {
    height: 58,
    width: 58,
    borderRadius: 29,
    borderWidth: StyleSheet.hairlineWidth,
    borderColor: '#EEE',
  },
};
