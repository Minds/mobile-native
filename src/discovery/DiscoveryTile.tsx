//@ts-nocheck
import React, { Component } from 'react';

import { Text, View, TouchableOpacity, StyleSheet } from 'react-native';

import FastImage from 'react-native-fast-image';

import { observer } from 'mobx-react';

import ExplicitOverlay from '../common/components/explicit/ExplicitOverlay';
import { CommonStyle as CS } from '../styles/Common';
import ThemedStyles from '../styles/ThemedStyles';
import ConnectivityAwareSmartImage from '../common/components/ConnectivityAwareSmartImage';

@observer
class DiscoveryTile extends Component {
  state = {
    error: false,
    style: null,
  };

  /**
   * Derive state from props
   * @param {object} nextProps
   * @param {object} prevState
   */
  static getDerivedStateFromProps(nextProps, prevState) {
    if (prevState.size !== nextProps.size) {
      return {
        style: { width: nextProps.size, height: nextProps.size },
      };
    }

    return null;
  }

  /**
   * On press
   */
  _onPress = () => {
    if (this.props.onPress) {
      this.props.onPress(this.props.entity);
    }
  };

  setError = () => {
    this.setState({ error: true });
  };

  setActive = () => {
    // bubble event up
    this.props.onLoadEnd && this.props.onLoadEnd();
  };

  /**
   * Render
   */
  render() {
    const entity = this.props.entity;

    // this optimization have some issues with the changes of the video auto-pause
    // if (!entity.is_visible) {
    //   return null;
    // }

    const url = entity.getThumbSource();

    // load gif with lower priority
    if (entity.isGif()) {
      url.priority = FastImage.priority.low;
    }

    const show_overlay = entity.shouldBeBlured();

    const overlay = show_overlay ? (
      <ExplicitOverlay entity={entity} iconSize={45} hideText={true} />
    ) : null;

    const boundary = this.props.boundaryText ? (
      <View
        style={[
          CS.positionAbsoluteTop,
          ThemedStyles.style.backgroundSeparator,
          CS.centered,
          styles.boundary,
        ]}>
        <Text>{this.props.boundaryText}</Text>
      </View>
    ) : null;

    return (
      <TouchableOpacity
        onPress={this._onPress}
        style={[this.state.style, styles.tile]}>
        <View
          style={[CS.flexContainer, ThemedStyles.style.backgroundSeparator]}>
          {boundary}
          <ConnectivityAwareSmartImage
            source={url}
            style={CS.positionAbsolute}
            onLoadEnd={this.setActive}
            onError={this.setError}
          />
          {overlay}
        </View>
      </TouchableOpacity>
    );
  }
}

export default DiscoveryTile;

const styles = StyleSheet.create({
  boundary: {
    height: 20,
    width: '100%',
    zIndex: 1000,
  },
  tile: {
    paddingTop: 1,
    paddingBottom: 1,
    paddingRight: 1,
    paddingLeft: 1,
  },
});
