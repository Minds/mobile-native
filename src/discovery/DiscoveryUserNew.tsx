//@ts-nocheck
import React, { Component } from 'react';

import { TouchableOpacity, Image, Keyboard, Text, View } from 'react-native';

import { observer } from 'mobx-react';

import { MINDS_CDN_URI } from '../config/Config';

// import { CommonStyle as CS } from '../styles/Common';
import { FLAG_SUBSCRIBE, FLAG_VIEW } from '../common/Permissions';
import SubscriptionButtonNew from '../channel/subscription/SubscriptionButtonNew';
import ThemedStyles from '../styles/ThemedStyles';
import type UserModel from '../channel/UserModel';

type PropsType = {
  row: {
    index: number;
    item: UserModel;
  };
  navigation: any;
  onUserTap?: Function;
};

@observer
class DiscoveryUser extends Component<PropsType> {
  /**
   * State
   */
  state = {
    guid: null,
  };

  /**
   * Derive state from props
   * @param {object} nextProps
   * @param {object} prevState
   */
  static getDerivedStateFromProps(nextProps, prevState) {
    if (prevState.guid !== nextProps.row.item.guid) {
      return {
        guid: nextProps.row.item.guid,
        source: {
          uri: MINDS_CDN_URI + 'icon/' + nextProps.row.item.guid + '/medium',
        },
      };
    }

    return null;
  }

  /**
   * Navigate To channel
   */
  _navToChannel = () => {
    Keyboard.dismiss();
    if (this.props.onUserTap) {
      this.props.onUserTap(this.props.row.item);
    }
    if (this.props.navigation) {
      if (
        this.props.row.item.isOpen() &&
        !this.props.row.item.can(FLAG_VIEW, true)
      ) {
        return;
      }
      this.props.navigation.push('Channel', { entity: this.props.row.item });
    }
  };

  /**
   * Render right button
   */
  renderRightButton() {
    const channel = this.props.row.item;

    if (
      channel.isOwner() ||
      this.props.hideButtons ||
      (channel.isOpen() && !channel.can(FLAG_SUBSCRIBE))
    ) {
      return;
    }

    const testID = this.props.testID
      ? `${this.props.testID}SubscriptionButton`
      : 'subscriptionButton';

    return <SubscriptionButtonNew channel={channel} testID={testID} />;
  }

  /**
   * Render
   */
  render() {
    const CS = ThemedStyles.style;
    const { row, subscribe, ...otherProps } = this.props;
    const renderRightButton = !(subscribe === false);
    return (
      <TouchableOpacity
        style={styles.row}
        onPress={this._navToChannel}
        {...otherProps}>
        <Image source={this.state.source} style={styles.avatar} />
        <View style={CS.flexContainerCenter}>
          <Text style={[styles.body, styles.title, CS.colorPrimaryText]}>
            {row.item.name}
          </Text>
          <Text style={[styles.body, styles.subtitle, CS.colorSecondaryText]}>
            @{row.item.username}
          </Text>
        </View>
        {renderRightButton && this.renderRightButton()}
      </TouchableOpacity>
    );
  }
}

export default DiscoveryUser;

const styles = {
  row: {
    flexDirection: 'row',
    alignItems: 'center',
    flexWrap: 'wrap',
    paddingTop: 10,
    paddingLeft: 12,
    paddingBottom: 10,
    paddingRight: 12,
  },
  body: {
    marginLeft: 16,
    height: 22,
    // flex: 1,
  },
  title: {
    fontSize: 17,
    fontWeight: '500',
  },
  subtitle: {
    fontSize: 14,
  },
  avatar: {
    height: 58,
    width: 58,
    borderRadius: 29,
  },
};
