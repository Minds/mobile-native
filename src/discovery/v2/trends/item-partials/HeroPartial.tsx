import React from 'react';
import { View } from 'react-native';
import RichPartialThumbnail from './RichPartialThumbnail';
import excerpt from '../../../../common/helpers/excerpt';
import {
  DISCOVERY_TRENDING_MAX_LENGTH,
  styles,
} from '../DiscoveryTrendsListItem';
import i18n from '../../../../common/services/i18n.service';
import MText from '../../../../common/components/MText';

type PropsType = {
  data: any;
};

const HeroPartial = ({ data }: PropsType) => {
  const entity = data.entity;
  return (
    <View style={[styles.heroContainer]}>
      <RichPartialThumbnail entity={entity} isHero />
      <View style={styles.container}>
        <View style={styles.body}>
          <MText style={styles.title}>
            {excerpt(data.title, DISCOVERY_TRENDING_MAX_LENGTH)}
          </MText>
          <MText style={styles.secondaryInformationBottom}>
            {data.volume} channels discussing -{' '}
            {i18n.date(parseInt(entity.time_created, 10) * 1000, 'friendly')}
          </MText>
        </View>
      </View>
    </View>
  );
};

export default HeroPartial;
