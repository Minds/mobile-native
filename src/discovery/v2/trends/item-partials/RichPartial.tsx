import React from 'react';
import { View, Text } from 'react-native';
import RichPartialThumbnail from './RichPartialThumbnail';
import excerpt from '../../../../common/helpers/excerpt';
import {
  DISCOVERY_TRENDING_MAX_LENGTH,
  styles,
} from '../DiscoveryTrendsListItem';
import i18n from '../../../../common/services/i18n.service';

type PropsType = {
  data: any;
};

const RichPartial = ({ data }: PropsType) => {
  const entity = data.entity;
  return (
    <View style={styles.container}>
      <View style={styles.body}>
        <Text style={styles.title}>
          {excerpt(data.title, DISCOVERY_TRENDING_MAX_LENGTH)}
        </Text>
        <Text style={styles.secondaryInformationBottom}>
          {data.volume} channels discussing -{' '}
          {i18n.date(parseInt(entity.time_created, 10) * 1000, 'friendly')}
        </Text>
      </View>
      <RichPartialThumbnail entity={data.entity} />
    </View>
  );
};

export default RichPartial;
