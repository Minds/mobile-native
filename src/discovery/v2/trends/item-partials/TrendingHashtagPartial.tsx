import React from 'react';
import { View } from 'react-native';
import ThemedStyles from '../../../../styles/ThemedStyles';
import { styles } from '../DiscoveryTrendsListItem';
import { Icon } from 'react-native-elements';
import MText from '../../../../common/components/MText';

type PropsType = {
  data: any;
};

const TrendingHashtagPartial = ({ data }: PropsType) => {
  return (
    <View style={styles.container}>
      <View style={styles.body}>
        <MText style={styles.secondaryInformationTop}>
          Trending {data.period}h - {data.volume} posts
        </MText>
        <MText style={styles.title}>#{data.hashtag}</MText>
      </View>
      <Icon
        type="material-community"
        color={ThemedStyles.getColor('TertiaryText')}
        name="chevron-right"
        size={32}
        style={styles.centered}
      />
    </View>
  );
};

export default TrendingHashtagPartial;
