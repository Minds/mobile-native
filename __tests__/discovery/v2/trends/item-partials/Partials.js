import 'react-native';
import React from 'react';
import { render } from '@testing-library/react-native';

import TrendingHashtagPartial from '../../../../../src/discovery/v2/trends/item-partials/TrendingHashtagPartial';
import RichPartial from '../../../../../src/discovery/v2/trends/item-partials/RichPartial';
import HeroPartial from '../../../../../src/discovery/v2/trends/item-partials/HeroPartial';
import { TrendingActivity } from '../../../../../__mocks__/fake/discovery/TrendingActivity';

const data = TrendingActivity;

describe('Partials tests', () => {
  it('renders correctly TrendingHashtagPartial', () => {
    const { toJSON } = render(<TrendingHashtagPartial data={data} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders correctly RichPartial', () => {
    const { toJSON } = render(<RichPartial data={data} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders correctly HeroPartial', () => {
    const { toJSON } = render(<HeroPartial data={data} />);
    expect(toJSON()).toMatchSnapshot();
  });
});
