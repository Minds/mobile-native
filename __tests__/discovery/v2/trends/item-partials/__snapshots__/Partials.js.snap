// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Partials tests renders correctly HeroPartial 1`] = `
<View
  style={
    Array [
      Array [
        Object {
          "flexDirection": "column",
          "paddingHorizontal": 0,
          "paddingTop": 0,
        },
      ],
    ]
  }
>
  <View
    style={
      Array [
        Array [
          Object {
            "height": 100,
            "width": 100,
          },
        ],
        Array [
          Object {
            "height": 444.6666666666667,
            "width": "100%",
          },
        ],
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Array [
            Array [
              Object {
                "height": 100,
                "width": 100,
              },
            ],
            Array [
              Object {
                "height": 444.6666666666667,
                "width": "100%",
              },
            ],
          ],
        ]
      }
    >
      <FastImageView
        onFastImageError={[Function]}
        onFastImageLoadEnd={[Function]}
        onFastImageProgress={[Function]}
        resizeMode="cover"
        retry={2}
        source={
          Object {
            "uri": "https://cdn.minds.com/fs/v1/thumbnail/1273869385403994122/xlarge/",
          }
        }
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderBottomWidth": 1,
          "display": "flex",
          "flexDirection": "row",
        },
        Object {
          "padding": 16,
        },
        Object {
          "borderColor": "#D3DBE3",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "flex": 1,
            "paddingRight": 10,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#43434D",
            },
            Object {
              "fontFamily": "Roboto",
            },
            Array [
              Object {
                "fontSize": 16,
                "fontWeight": "bold",
              },
            ],
          ]
        }
      >
        Wow, what do we call this? It's beautiful, isn't it?
#Nature
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#43434D",
            },
            Object {
              "fontFamily": "Roboto",
            },
            Array [
              Object {
                "paddingTop": 8,
              },
              Object {
                "color": "#72727C",
              },
            ],
          ]
        }
      >
        5
         channels discussing -
         
        1h ago
      </Text>
    </View>
  </View>
</View>
`;

exports[`Partials tests renders correctly RichPartial 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderBottomWidth": 1,
        "display": "flex",
        "flexDirection": "row",
      },
      Object {
        "padding": 16,
      },
      Object {
        "borderColor": "#D3DBE3",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flex": 1,
          "paddingRight": 10,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#43434D",
          },
          Object {
            "fontFamily": "Roboto",
          },
          Array [
            Object {
              "fontSize": 16,
              "fontWeight": "bold",
            },
          ],
        ]
      }
    >
      Wow, what do we call this? It's beautiful, isn't it?
#Nature
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#43434D",
          },
          Object {
            "fontFamily": "Roboto",
          },
          Array [
            Object {
              "paddingTop": 8,
            },
            Object {
              "color": "#72727C",
            },
          ],
        ]
      }
    >
      5
       channels discussing -
       
      1h ago
    </Text>
  </View>
  <View
    style={
      Array [
        Array [
          Object {
            "height": 100,
            "width": 100,
          },
        ],
        null,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Array [
            Array [
              Object {
                "height": 100,
                "width": 100,
              },
            ],
            null,
          ],
        ]
      }
    >
      <FastImageView
        onFastImageError={[Function]}
        onFastImageLoadEnd={[Function]}
        onFastImageProgress={[Function]}
        resizeMode="cover"
        retry={2}
        source={
          Object {
            "uri": "https://cdn.minds.com/fs/v1/thumbnail/1273869385403994122/xlarge/",
          }
        }
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[`Partials tests renders correctly TrendingHashtagPartial 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderBottomWidth": 1,
        "display": "flex",
        "flexDirection": "row",
      },
      Object {
        "padding": 16,
      },
      Object {
        "borderColor": "#D3DBE3",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flex": 1,
          "paddingRight": 10,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#43434D",
          },
          Object {
            "fontFamily": "Roboto",
          },
          Array [
            Object {
              "paddingBottom": 8,
            },
            Object {
              "color": "#72727C",
            },
          ],
        ]
      }
    >
      Trending 
      9.224444444444444
      h - 
      5
       posts
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#43434D",
          },
          Object {
            "fontFamily": "Roboto",
          },
          Array [
            Object {
              "fontSize": 16,
              "fontWeight": "bold",
            },
          ],
        ]
      }
    >
      #
      nature
    </Text>
  </View>
  <View
    style={
      Object {
        "overflow": "hidden",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignSelf": "center",
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "justifyContent": "center",
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#9B9B9B",
                "fontSize": 32,
              },
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "fontFamily": "Material Design Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
          testID="iconIcon"
        >
          ó°…‚
        </Text>
      </View>
    </View>
  </View>
</View>
`;
